share/lib/python/neuron/rxd/geometry3d/CMakeLists.txt

```diff
@@ -0,0 +1,40 @@
+add_compile_options(${NRN_COMPILE_FLAGS})
+add_compile_definitions(${NRN_COMPILE_DEFS})
+add_link_options(${NRN_LINK_FLAGS})
+
+set(SOURCES graphicsPrimitives ctng surfaces)
+
+foreach(source ${SOURCES})
+  add_custom_command(
+    OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/${source}.cpp
+    COMMAND ${PYTHON_EXECUTABLE} -m cython --cplus "${CMAKE_CURRENT_SOURCE_DIR}/${source}.pyx"
+            --output-file ${CMAKE_CURRENT_BINARY_DIR}/${source}.cpp
+    DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/${source}.pyx
+    VERBATIM)
+endforeach()
+
+add_library(graphicsPrimitives SHARED ${CMAKE_CURRENT_BINARY_DIR}/graphicsPrimitives.cpp)
+
+add_library(ctng SHARED ${CMAKE_CURRENT_BINARY_DIR}/ctng.cpp)
+
+target_include_directories(ctng PUBLIC "${NRN_PYTHON_INCLUDES}" "${NRN_PYTHON_NUMPY_INCLUDES}")
+
+add_library(
+  surfaces SHARED
+  ${CMAKE_CURRENT_BINARY_DIR}/surfaces.cpp
+  ${PROJECT_SOURCE_DIR}/src/nrnpython/rxd_marching_cubes.cpp
+  ${PROJECT_SOURCE_DIR}/src/nrnpython/rxd_llgramarea.cpp)
+
+target_include_directories(surfaces PUBLIC "${NRN_PYTHON_INCLUDES}" "${NRN_PYTHON_NUMPY_INCLUDES}")
+
+foreach(source ${SOURCES})
+  target_link_libraries(${source} nrniv_lib rxdmath)
+  if(NRN_ENABLE_PYTHON_DYNAMIC)
+    foreach(val RANGE ${NRN_PYTHON_ITERATION_LIMIT})
+      list(GET NRN_PYTHON_VERSIONS ${val} pyver)
+      target_link_libraries(${source} nrnpython${pyver})
+    endforeach()
+    # else() target_link_libraries(${source} ${NRN_DEFAULT_PYTHON_LIBRARIES})
+  endif()
+  install(TARGETS ${source} DESTINATION ${NRN_INSTALL_SHARE_LIB_DIR})
+endforeach()
```

Comment:
@JCGoran: I see this is coming in full circle 😅  i.e. not a long time ago, the rxd-related building part was controlled separately (e.g. in #2235, it was separate `setup.py` and then partly controlled via CMake).

IIRC, the reason for moving that into top-level `setup.py` was to make those extensions as first-class citizen i.e. handle those "naturally".  Maybe @ferdonline can better explain this but just want to point out that it was partly like this before. So it would be good to understand why we are going back to the old style.

Reply:
Oh haha I actually had to backport that PR to 8.2.4, I guess I didn't understand its full extent back then 😅 The main idea behind this PR is to regain full control over the build procedure (Cython's `Extension` is old, and uses setuptools, which is find for pure Python modules, but very cumbersome and inflexible to use for binary extensions like our usecase here), and consequently use only one build system (in this case, CMake). This would allow us to eventually migrate to a static config (i.e. no more dynamic `setup.py`), and use a build system (like `scikit-build-core`) that just calls CMake under the hood. This way, we would only need to maintain one build system, and use the Python one as a very thin wrapper around it.

Reply:
> use a build system (like scikit-build-core) that just calls CMake under the hood

I read until I arrived at `scikit-build` 😅 (old annoyance with the maintenance aspects of this package but I see that `-core` is recent and actively developed).

I will let @ferdonline provide his opinion.